R/data.table.R

```diff
@@ -2463,6 +2463,9 @@ split.data.table = function(x, f, drop = FALSE, by, sorted = FALSE, keep.by = TR
     if (!missing(by))
       stopf("passing 'f' argument together with 'by' is not allowed, use 'by' when split by column in data.table and 'f' when split by external factor")
     # same as split.data.frame - handling all exceptions, factor orders etc, in a single stream of processing was a nightmare in factor and drop consistency
+    # evaluate formula mirroring split.data.frame #5392. Mimics base::.formula2varlist.
+    if (inherits(f, "formula"))
```

Comment:
`split.data.frame()` will make some noise for improper formula input such as `f=x~y+z`, shall we do similar validation?

Reply:
If we want to do this then we should probably just dispatch to `split.data.frame` in case of `formula`?

Reply:
The workaround to an R dependency is to copy-paste the `.formula2varlist` definition. But that function is larger than I'd like it to be for a simple copy-paste job, so still favor deferring this until requested.